### PR TITLE
[WebNN] Support steps >= 1 for slice operator

### PR DESCRIPTION
**Pull request by shiyi9801**

### CODE REVIEW

**Review by guschmue — APPROVED** [review]

File: onnxruntime/core/providers/webnn/builders/impl/slice_op_builder.cc [hunk 1]
```diff
@@ -144,18 +148,19 @@ bool SliceOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers,
...
-    // WebNN doesn't support steps other than 1.
+    // WebNN doesn't support steps less than 1.
     if (data_type == ONNX_NAMESPACE::TensorProto_DataType_INT64) {
-      if (!std::all_of(reinterpret_cast<int64_t*>(unpacked_tensor.data()),
-                       reinterpret_cast<int64_t*>(unpacked_tensor.data() + unpacked_tensor.size()),
-                       [](int64_t i) { return i == 1; })) {
+      if (std::any_of(reinterpret_cast<int64_t*>(unpacked_tensor.data()),
+                      reinterpret_cast<int64_t*>(unpacked_tensor.data() + unpacked_tensor.size()),
+                      [](int64_t i) { return i < 1; })) {
+        LOGS(logger, VERBOSE) << "WebNN slice doesn's support steps less than 1";
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/builders/impl/slice_op_builder.cc:156]:
```suggestion
        LOGS(logger, VERBOSE) << "WebNN slice doesn't support steps less than 1";
```

File: onnxruntime/core/providers/webnn/builders/impl/slice_op_builder.cc [hunk 2]
```diff
@@ -144,18 +148,19 @@ bool SliceOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers,
...
       }
     } else if (data_type == ONNX_NAMESPACE::TensorProto_DataType_INT32) {
-      if (!std::all_of(reinterpret_cast<int32_t*>(unpacked_tensor.data()),
-                       reinterpret_cast<int32_t*>(unpacked_tensor.data()) +
-                           unpacked_tensor.size() / sizeof(int32_t),
-                       [](int32_t i) { return i == 1; })) {
+      if (std::any_of(reinterpret_cast<int32_t*>(unpacked_tensor.data()),
+                      reinterpret_cast<int32_t*>(unpacked_tensor.data()) + unpacked_tensor.size() / sizeof(int32_t),
+                      [](int32_t i) { return i < 1; })) {
+        LOGS(logger, VERBOSE) << "WebNN slice doesn's support steps less than 1";
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/builders/impl/slice_op_builder.cc:163]:
```suggestion
        LOGS(logger, VERBOSE) << "WebNN slice doesn't support steps less than 1";
```

**Comment by shiyi9801** [comment]
@guschmue @fdwr PTAL, thanks!

**Review by fdwr — APPROVED** [review]
👍

**Pull request merged**